File: Csharp/createCategory.cs
using Newtonsoft.Json.Linq;
using System;
using System.IO;
using System.Net;
using System.Xml;

namespace createCategory
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            // Changez le Path pour correspondre à la destination de votre fichier de configuration
            var doc = new XmlDocument();
            doc.Load("config.xml");

            var xKey = doc.DocumentElement.SelectSingleNode("/config/xKey").InnerText;
            var baseUrl = doc.DocumentElement.SelectSingleNode("/config/url").InnerText;

            string categoriesId = null;	// Id de la categorie a modifier ('null' si la categorie doit etre cree)

            var name = "Category Csharp";	// Nom de la categorie
            var description = "Category (Csharp)";	// Description de la categorie


            // On trouve l'adresse pour la requete
            var url = baseUrl + "categories/" + categoriesId;

            Console.Write(xKey + "\n");

            // Creation du Json du message
            var jsonData = new JObject();
            jsonData.Add("name", name);
            jsonData.Add("description", description);

            // On affiche le message
            Console.Write(jsonData + "\n");


            // Lancement de la connexion pour remplir la requete
            var connection = Utils.allConnection(url, xKey, jsonData, (segmentId == null ? "POST" : "PUT"));
            
            // Verification des reponses
            if ((int)connection[0] != 200)
            {
                // Affichage de l'erreur
                Console.Write("\nError : " + connection[0]);
            }
            else
            {
                // La categorie a bien ete cree
                Console.Write("The category  : " + name + " is created / update.");
            }
            // Attente de lecture (optionnel)
            Console.ReadLine();

            ((HttpWebResponse)connection[2]).Close();
        }
    }
}

File: python/CreateCategory.py
"""
Ce script permet de créer une Catégorie.
"""

import json
import requests
import Utils

def create_category(config):
    """
    Création d'une Catégorie

    Pour plus d'informations : https://backoffice.mailperformance.com/doc/#api-Category-CreateCategory

    Description du fichier createCategory.json

    name                => Le nom de la catégorie
    description         => La description de la catégorie

    """

    with open('Models/createCategory.json') as data_file:
        data = json.load(data_file)

    data_json = json.JSONEncoder().encode(data)

    print("Creation of the category.")
    url = config['API']['url'] + 'categories/'
    headers = Utils.create_headers(config['API']['xKey'], len(data_json))
    req = requests.post(url, data=data_json, headers=headers)
    req.raise_for_status()

    print("The category has been Created.")

def main():
    config = Utils.load_config()
    create_category(config)

if __name__ == '__main__':
    main()
